pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/IndexService.java

```diff
@@ -53,18 +56,42 @@
  */
 @ThreadSafe
 public class IndexService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(IndexService.class);
 
   private static volatile IndexService _instance = fromServiceLoader();
 
   private final List<IndexType<?, ?, ?>> _allIndexes;
   private final Map<String, IndexType<?, ?, ?>> _allIndexesById;
 
-  private IndexService(Set<IndexPlugin<?>> allPlugins) {
-    ImmutableMap.Builder<String, IndexType<?, ?, ?>> builder = ImmutableMap.builder();
+  public IndexService(Set<IndexPlugin<?>> allPlugins) {
```

Comment:
(minor) Is this exposed just for testing purpose? If so, I'd suggest making it package private, and add `@VisibleForTesting`

Reply:
I've opened this method so anyone can register their own IndexService by calling the static method IndexService.setInstance, which was open before but useless given nobody could instanciate an IndexService different than the one that reads from the classpath